drop-storage/src/lib.rs

```diff
@@ -488,8 +488,11 @@ impl Storage {
         let task = async {
             let conn = self.conn.lock().await;
             conn.execute(
-                "INSERT INTO outgoing_path_pending_states (path_id) VALUES ((SELECT id FROM \
-                 outgoing_paths WHERE transfer_id = ?1 AND path_hash = ?2))",
+                r#"
+                INSERT INTO outgoing_path_pending_states (path_id)
+                SELECT id
+                FROM outgoing_paths WHERE transfer_id = ?1 AND path_hash = ?2
+                "#,
```

Comment:
Technically these queries might insert 0..N values, maybe it's worth asserting that we're expecting just 0 or 1?

Reply:
Do you mean LIMIT 1 ?

Reply:
That's a good suggestion as well. I was concerned that we expect tat it should always be 1 item, then we found a race that it might be 0 which showed to us that there's an actual race condition in the codebase. Doing `LIMIT 1` would hide the fact that due to some mistake there are more than 1 entry.

Reply:
How do you want to achieve this?

Reply:
I don't know how to handle this. Leaving as it is